pkg/chain/ethereum/ethutil/rate_limiter_test.go

```diff
@@ -0,0 +1,246 @@
+package ethutil
+
+import (
+	"context"
+	"fmt"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"math/big"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+)
+
+const requestDuration = 250 * time.Millisecond
+
+func TestRateLimiter(t *testing.T) {
+	requestsPerSecondLimit := 200
+	concurrencyLimit := 50
+	acquirePermitTimeout := time.Minute
+	requests := 1000
+
+	backend := &mockBackend{make([]string, 0), sync.Mutex{}}
+
+	rateLimitingBackend := WrapRateLimiting(
+		backend,
+		&RateLimiterConfig{
+			RequestsPerSecondLimit: requestsPerSecondLimit,
+			ConcurrencyLimit:       concurrencyLimit,
+			AcquirePermitTimeout:   acquirePermitTimeout,
+		},
+	)
+
+	wg := sync.WaitGroup{}
+	wg.Add(requests)
+
+	startSignal := make(chan struct{})
+
+	for i := 0; i < requests; i++ {
+		go func() {
+			<-startSignal
+
+			err := rateLimitingBackend.SendTransaction(context.Background(), nil)
+			if err != nil {
+				t.Errorf("unexpected error: [%v]", err)
+			}
+
+			wg.Done()
+		}()
+	}
+
+	startTime := time.Now()
+	close(startSignal)
+
+	wg.Wait()
+
+	duration := time.Now().Sub(startTime)
+	averageRequestsPerSecond := float64(requests) / duration.Seconds()
+
+	if averageRequestsPerSecond > float64(requestsPerSecondLimit) {
+		t.Errorf(
+			"average requests per second exceeded the limit\n"+
+				"limit:  [%v]\n"+
+				"actual: [%v]",
+			requestsPerSecondLimit,
+			averageRequestsPerSecond,
+		)
+	}
```

Comment:
This test takes now 5 seconds to complete and we are covering just `SendTransaction`. I think we want to confirm that all functions of our decorator are covered. I'd lower `requestDuration`, `requestsPerSecondLimit`, and/or `requests` and make sure all functions are covered.